### PR TITLE
react-list-cmpc

### PR DESCRIPTION
**Pull request by ema999**
## Summary

Adding a new collapsable list with remove option from CMPC project

## Screenshots

https://user-images.githubusercontent.com/5349650/117709497-61005a80-b1a7-11eb-92e8-40a3db99e855.mov

## Tests

image
image

Note: There's only one small "bug", which is that when the content's height is dinamically changed after resizing the screen and the resulting height is higher (this can happen when you increase the screen width) the collapsible element does not shrink as it should. Nothing breaks, but it's not the best way to show the content. Given that I couldn figure out a solution, for the time being it works like this, since this use case is very rare. Below I leave an example of this:

https://user-images.githubusercontent.com/5349650/117709901-dbc97580-b1a7-11eb-9128-2d9364d444f4.mov

### CODE REVIEW

File: cookbook-react/src/recipes/lists/list-cmpc/index.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+function List({ title, items, handleDelete }: ListProps) {
+  const [itemsList, setItems] = useState(items);
+
+  if (!itemsList?.length) {
+    return null;
+  }
+
+  const onDelete = (id: number) => {
+    const itemsfiltered: Item[] = itemsList.filter((item: Item) => item.id !== id);
```
Comment (damfinkel) [inline cookbook-react/src/recipes/lists/list-cmpc/index.tsx:25]: `filteredItems`

In english, the noun modifier is always before the noun itself, for example: `black cat`, `big house`, `modified element`, `filtered items`

File: cookbook-react/src/recipes/lists/list-cmpc/index.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            <span className={styles.textNormal}>{item.name}</span>
+            <div className={`relative pointer ${styles.closeIcon}`} onClick={() => onDelete(item.id)} />
+          </li>
+        ))}
+      </ul>
+    </div>
+  );
+}
+
+List.defaultProps = {
```
Comment (damfinkel) [inline cookbook-react/src/recipes/lists/list-cmpc/index.tsx:49]: Add here as well the comment that this is only for example purposes and to remove it in a real project.

File: cookbook-react/src/recipes/lists/list-cmpc/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+.text-normal {
+  color: $black;
+  font-weight: 700;
+  font-size: 16px;
+  line-height: 1.2;
+}
+
+.close-icon {
```
Comment (damfinkel) [inline cookbook-react/src/recipes/lists/list-cmpc/styles.module.scss:32]: In the image it seems like the close icon is not centered vertically, please check that

File: cookbook-react/src/recipes/lists/list-cmpc/styles.module.scss [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  &:not(:last-child) {
+    margin-bottom: 12px;
+  }
+}
+
+.text-normal {
+  color: $black;
+  font-weight: 700;
```
Comment (damfinkel) [inline cookbook-react/src/recipes/lists/list-cmpc/styles.module.scss:27]: sort properties

File: cookbook-react/src/recipes/lists/list-cmpc/styles.module.scss [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  font-size: 16px;
+  line-height: 1.2;
+}
+
+.close-icon {
+  cursor: pointer;
+  height: 20px;
+  width: 20px;
```
Comment (damfinkel) [inline cookbook-react/src/recipes/lists/list-cmpc/styles.module.scss:35]: sort properties

File: cookbook-react/src/recipes/lists/list-cmpc/index.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
```
Comment (damfinkel) [inline cookbook-react/src/recipes/lists/list-cmpc/index.tsx:1]: Can we change the name of the folder to `cmpc list`?

**Review by damfinkel — CHANGES_REQUESTED** [review]

**Comment by alejobh** [comment]
tests please

**Review by alejobh — APPROVED** [review]

File: cookbook-react/src/hooks/useCollapse/index.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    const newHeight = scrollHeight ?? 0;
+    setOriginalHeight(newHeight);
+  }, []);
+
+  // Sets initial height values
+  useEffect(() => {
+    collapsibleRef.current!.style.transition = `height ${collapseTime}ms ease-out, padding ${collapseTime}ms ease-out, margin ${collapseTime}ms ease-out`;
+    collapsibleRef.current!.style.overflow = 'hidden';
+
+    calculateHeight();
```
Comment (pabloferro) [inline cookbook-react/src/hooks/useCollapse/index.tsx:46]:
```suggestion
    collapsibleRef.current!.style.transition = `height ${collapseTime}ms ease-out, padding ${collapseTime}ms ease-out, margin ${collapseTime}ms ease-out`;
    collapsibleRef.current!.style.overflow = 'hidden';

    calculateHeight();
```
What if current is not defined yet 🤔 , why don't you add an if?

File: cookbook-react/src/recipes/lists/valparaiso/Item/styles.module.scss [hunk 8]
```diff
@@ -0,0 +1,47 @@
+$mercury: #E5E5E5;
+$alabaster: #F7F7F7;
+$oslo-gray: #92978B;
+$black: #000;
```
Comment (pabloferro) [inline cookbook-react/src/recipes/lists/valparaiso/Item/styles.module.scss:4]: I'd use constants with semantic meaning, like primary, disabled, background, etc.

File: cookbook-react/src/recipes/lists/valparaiso/List/styles.module.scss [hunk 9]
```diff
@@ -0,0 +1,29 @@
+$alabaster-tertiary: #DFDFDF;
+$alabaster: #F7F7F7;
+$black: #000;
```
Comment (pabloferro) [inline cookbook-react/src/recipes/lists/valparaiso/List/styles.module.scss:3]: Same here 😄

File: cookbook-react/src/recipes/lists/valparaiso/List/styles.module.scss [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+.list-item {
+  border-radius: 6px;
+  height: 50px;
+  padding: 20px 20px 20px 25px;
```
Comment (pabloferro) [inline cookbook-react/src/recipes/lists/valparaiso/List/styles.module.scss:8]: _suggestion: I think it's clearer._
```suggestion
  padding: 20px;
  padding-left: 25px;
```

**Review by pabloferro — CHANGES_REQUESTED** [review]

**Review by pabloferro — APPROVED** [review]

**Pull request merged**